library/ssl_tls13_client.c

```diff
@@ -1335,6 +1335,53 @@ static int ssl_tls13_is_downgrade_negotiation( mbedtls_ssl_context *ssl,
     return( 0 );
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+/*
+ * ssl_tls13_parse_ee_early_data_ext()
+ *      Parse early data indication extension in EncryptedExtensions.
+ *
+ * struct {} Empty;
+ *
+ * struct {
+ *   select (Handshake.msg_type) {
+ *     ...
+ *     case client_hello:         Empty;
+ *     case encrypted_extensions: Empty;
+ *   };
+ * } EarlyDataIndication;
+ *
+ */
+
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_parse_ee_early_data_ext( mbedtls_ssl_context *ssl,
+                                              const unsigned char *buf,
+                                              size_t len )
+{
+    if( ssl->early_data_status < MBEDTLS_SSL_EARLY_DATA_STATUS_INDICATION_SENT )
+    {
+        /* The server must not send the EarlyDataIndication if the
+         * client hasn't indicated the use of early data. */
+        MBEDTLS_SSL_PEND_FATAL_ALERT( MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER,
+                                      MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER );
+        return( MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER );
+    }
+
+    if( len != 0 )
+    {
+        /* The message must be empty. */
+        MBEDTLS_SSL_PEND_FATAL_ALERT( MBEDTLS_SSL_ALERT_MSG_DECODE_ERROR,
+                                      MBEDTLS_ERR_SSL_DECODE_ERROR );
+        return( MBEDTLS_ERR_SSL_DECODE_ERROR );
+    }
+
+    /* Nothing to parse */
+    ((void) buf);
+
+    ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED;
```

Comment:
```suggestion
    if( ssl->handshake->received_extension & MBEDTLS_SSL_EXT_MASK( EARLY_DATA ) )
        ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED;
    else
        ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_REJECT;
```

Reply:
I didn't find the function postprocess_encrypted_extesion(), so I put **ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED;** in the ssl_tls13_parse_encrypted_extensions().

Reply:
line 2132.
Not Addressed

Reply:
I am not sure why we should put the status change in line 2132. 
Seems it's more readable and directly if we put the status change **"ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED;**" just follow up the early data extension here. 
And, we have no need of the **if( ssl->handshake->received_extension & MBEDTLS_SSL_EXT_MASK( EARLY_DATA ) )**, which can save the code size.